publisher-provider/publisher-provider-app/src/main/java/nl/idgis/publisher/provider/database/Database.java

```diff
@@ -19,7 +24,29 @@ public static Props props(Config config, String name) {
 	}
 	
 	@Override
-	protected Props createTransaction(Connection connection) {
-		return DatabaseTransaction.props(config, connection);
+	protected Props createTransaction(Connection connection) throws ConfigException {
+        DatabaseType databaseVendor;
+
+		// Value should come from config
+		String vendor = config.getString("vendor");
+		log.debug(String.format("Using database: %s", vendor));
+
+        try {
+            databaseVendor = DatabaseType.valueOf(config.getString("vendor").toUpperCase());
+        } catch(IllegalArgumentException iae) {
+            throw new ConfigException.BadValue("database {vendor}", "Invalid vendor supplied in config");
+        }
+
+
+		/* Return props from Oracle or postgres */
+        switch(databaseVendor) {
+            case ORACLE:
+                return OracleDatabaseTransaction.props(config, connection);
+            case POSTGRES:
+                return PostgresDatabaseTransaction.props(config, connection);
+            default:
+                log.error("Vendor is not supported");
+                throw new ConfigException.BadValue("database {vendor}", "Invalid vendor supplied in config");

```

Comment:
Beter om hier in de message van de exceptie terug te laten komen dat de vendor waarde welliswaar bekend is (lees: komt voor in de enum) maar dat deze in deze context niet ondersteund wordt. Voorstel: 'Invalid' -> 'Unsupported'

Reply:
Opgelost